api/controllers/game/rematch.js

```diff
@@ -40,14 +40,38 @@ module.exports = async function (req, res) {
       shouldNewGameBeRanked,
       gameService.GameStatus.STARTED,
     );
-    await Promise.all([
+    const [ , , p0, p1] = await Promise.all([
       Game.updateOne({ id: updatedGame.id }).set({ rematchGame: newGame.id }),
       Game.replaceCollection(newGame.id, 'players').members([newP0Id, newP1Id]),
       User.updateOne({ id: newP0Id }).set({ pNum: 0 }),
       User.updateOne({ id: newP1Id }).set({ pNum: 1 }),
     ]);
     updatedGame.rematchGame = newGame.id;
 
+    //Get all exisiting rematchGames
+    const rematchGames = [];
+    const getRematchGames = async (gameId) => {
+      const gameToAdd = await Game.findOne({ rematchGame: gameId });
+      if (!gameToAdd) {
+        return;
+      }
+      rematchGames.unshift(gameToAdd);
+      if (!gameToAdd.rematchGame) {
+        return; 
+      }
+      await getRematchGames(gameToAdd.id);
+    };
+    await getRematchGames(newGame.id);
+    const seriesP0 = [p0, p1].find(({ id }) => id === rematchGames[0].p0);
+    const seriesP1 = [p0, p1].find(({ id }) => id === rematchGames[0].p1);
+    //Get rematchGame win counts
+    let player0Wins = rematchGames.filter(({winner}) => winner === seriesP0.id).length;
+    let player1wins = rematchGames.filter(({winner}) => winner === seriesP1.id).length;
+    let stalemates = rematchGames.filter(({winner}) => winner === null).length;
+    
+    const newName = `${seriesP0.username} VS ${seriesP1.username} ${player0Wins}-${player1wins}-${stalemates}`;
```

Comment:
@Haviles04 @seriouslysean what do you think is clearer?

Reply:
I kinda like how after the .find P0, and P1 are no longer used and the rest of it uses the same objects instead of jumping back and forth, but I think either way is pretty clear, and am happy with either